libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/KeyValueEncryptedFileStore.java

```diff
@@ -144,28 +160,13 @@ public boolean saveValue(String key, String value) {
             SmartStoreLogger.w(TAG, "saveValue: Invalid value supplied: null");
             return false;
         }
-        FileOutputStream f = null;
         try {
-            final File file = getFileForKey(key);
-            f = new FileOutputStream(file);
-            byte[] encryptedContent = Encryptor.encryptWithoutBase64Encoding(value.getBytes(), encryptionKey);
-            if (encryptedContent != null) {
-                f.write(encryptedContent);
-                return true;
-            } else {
-                return false;
-            }
-        } catch (IOException e) {
-            SmartStoreLogger.e(TAG, "IOException occurred while saving value to filesystem", e);
+            if (kvVersion >= 2) encryptStringToFile(getKeyFile(key), key, encryptionKey);
```

Comment:
High level design question: Instead of maintaining a version here that's separate from the Mobile SDK version, could we use `SmartStoreUpgradeManager` to perform a one-time upgrade step between `9.0` and `9.1` like we do with the other components? You'll get the version information for free since we already write the current version to a file, and apps will automatically be upgraded because upgrade steps are wired into the `init` call.

Reply:
We can't upgrade. In 9.0, we one-way hash the keys to produce the file names. There is no way to get the keys back.

Reply:
For the version number, I used https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/dev/libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java#L62 as inspiration.

Reply:
Gotcha. Is there a long term plan to remove the old code, i.e. `V1`?